server/main_server/src/main.cpp

```diff
@@ -1009,6 +1028,21 @@ Mandatory arguments to long options are mandatory for short options too.
         }
 
         log_server::info("{}: Agent Factory PID = [{}].", __func__, g_pid_af);
+
+        // Wait for the agent factory's signal that it has completed initialization. This effectively
+        // means the agent factory has opened its listening socket and is entering its main loop.
+        while (0 == g_agent_factory_initialized) {
+            std::this_thread::sleep_for(std::chrono::milliseconds{250});
+            log_server::debug("{}: Waiting for agent factory to complete initialization.", __func__);
```

Comment:
Suggesting flipping these since the message about waiting comes after the waiting has occurred :)
```suggestion
            log_server::debug("{}: Waiting for agent factory to complete initialization.", __func__);
            std::this_thread::sleep_for(std::chrono::milliseconds{250});
```

Reply:
Will do. That suggestion aligns with the rest of the code too.

Reply:
Done.